### DISCUSSION TITLE
Making custom plots

### DISCUSSION THREAD

**Discussion by Rajivrocks-Ltd** [trackmania-rl/tmrl#103] [Q&A] [answered]
Hello again, I wanted to make my own custom plots of the various losses I have and the returns values. The losses I can figure out, but how do you handle printing the returns in the trainer console? because if I could grab that I could simply save it in a numpy array and use that for plotting after training.

**Comment by yannbouteiller** [comment]
Hi,

If you are using wandb, they have a python API that enables you to retrieve everything from there.

Otherwise, the stats are retrieved in the code [here](https://github.com/trackmania-rl/tmrl/blob/master/tmrl%2Fnetworking.py#L328) (there is no API in `tmrl` for logging these stats locally ATM, so you'd have to write a custom version of `Trainer.run()` using lower-level methods of your `Trainer` object.)

**Reply by Rajivrocks-Ltd** [reply, marked as answer]
The losses are also in wandb? Because when I look at the charts I don't see them plotted? EDIT: Oh nevermind it should be there. I see when I exit my trainer it shows all the stats that are being logged.

**Reply by yannbouteiller** [reply]
For sure they are plotted, they're one of the most important metrics 😉

There is also an option for logging an insane amount of useless metrics somewhere, I was using that for understanding performance collapse in SAC before implementing [Mahmood et al.](https://openreview.net/forum?id=m9Jfdz4ymO) to alleviate this recurring issue.
